### PR TITLE
AIX facelift

### PR DESCRIPTION
**Pull request by dot-asm**
This doesn't change shared library naming yet (being discussed in #6368), only unifies targets and adds -bsymbolic and explicit export list for shared libraries.

### CODE REVIEW

**Comment by dot-asm** [comment]
I'm considering adding another unification commit, so please hold approvals...

**Comment by dot-asm** [comment]
Extra commit added...

File: Configurations/10-main.conf [hunk 1]
```diff
@@ -1114,85 +1114,74 @@ my %targets = (
     # variable, then you should know that in OpenSSL case it's considered
     # only in ./config. Once configured, build procedure remains "deaf" to
     # current value of $OBJECT_MODE.
-    "aix-gcc" => {
+    "aix-common" => {
         inherit_from     => [ "BASE_unix", asm("ppc32_asm") ],
```
Comment (levitte) [inline Configurations/10-main.conf]: The asm part should move to the 32-bit config targets.

Comment (dot-asm) [reply Configurations/10-main.conf]: Note that it was an add-on commit, i.e. it's not an oversight, but totally conscious choice. Idea is to exploit the fact that ppc64_asm is actually just an add-on to ppc32_asm, it adds couple of modules. Another idea is to reflect the fact that AIX is available only for PPC platform. Well, one can still argue that it belongs in specific targets, at least that's where perlasm_scheme is still specified... You can simply approve all but last commit then. Shall we say so?

Comment (dot-asm) [reply Configurations/10-main.conf]: I intend to use Rich's approval that I assume stands for all commits but last add-on...

Comment (levitte) [reply Configurations/10-main.conf]: That's sounds good enough for me

**Pull request closed without merging**